dynawo/sources/Modeler/DataInterface/DYNCriteria.cpp

```diff
@@ -148,34 +256,74 @@ LoadCriteria::checkCriteria(double t, bool finalStep) {
     }
   }
 
-  if (atLeastOneEligibleLoadWasFound && params_->getType() == criteria::CriteriaParams::SUM) {
-    if (params_->hasPMax() && sum > params_->getPMax()) {
-#ifdef _DEBUG_
-  for (size_t i = 0; i < sourcesAddedIntoSum.size(); ++i) {
-    Trace::info() << DYNLog(SourcePowerTakenIntoAccount, "load", sourcesAddedIntoSum[i]->getID(), params_->getId(),
-        sourcesAddedIntoSum[i]->getP(), sourcesAddedIntoSum[i]->getBusInterface()->getStateVarV()) << Trace::endline;
-  }
-#endif
-      Message mess = DYNLog(SourcePowerAboveMax, sum, params_->getPMax(), params_->getId());
-      Trace::info() << mess << Trace::endline;
-      failingCriteria_.push_back(std::make_pair(t, mess.str()));
-      return false;
+  if (params_->getType() == criteria::CriteriaParams::LOCAL_VALUE) {
+    if (!isCriteriaOk && timeline != nullptr) {
+      printAllFailingCriteriaIntoLog(distanceToLoadFailingCriteriaMap, timeline, t);
     }
-    if (params_->hasPMin() && sum < params_->getPMin()) {
-#ifdef _DEBUG_
-  for (size_t i = 0; i < sourcesAddedIntoSum.size(); ++i) {
-    Trace::info() << DYNLog(SourcePowerTakenIntoAccount, "load", sourcesAddedIntoSum[i]->getID(), params_->getId(),
-        sourcesAddedIntoSum[i]->getP(), sourcesAddedIntoSum[i]->getBusInterface()->getStateVarV()) << Trace::endline;
-  }
-#endif
-      Message mess = DYNLog(SourcePowerBelowMin, sum, params_->getPMin(), params_->getId());
-      Trace::info() << mess << Trace::endline;
-      failingCriteria_.push_back(std::make_pair(t, mess.str()));
-      return false;
+  } else if (params_->getType() == criteria::CriteriaParams::SUM) {
+    if (atLeastOneEligibleLoadWasFound) {
+      if (params_->hasPMax() && sum > params_->getPMax()) {
+        std::sort(sourcesAddedIntoSum.begin(),
+                  sourcesAddedIntoSum.end(),
+                  [](const boost::shared_ptr<LoadInterface>& load1, const boost::shared_ptr<LoadInterface>& load2) -> bool {
+                    return load1->getP() > load2->getP();
+                  });
+        for (size_t i = 0; i < sourcesAddedIntoSum.size(); ++i) {
+          Trace::debug() << DYNLog(SourcePowerTakenIntoAccount, "load", sourcesAddedIntoSum[i]->getID(), params_->getId(),
+              sourcesAddedIntoSum[i]->getP(), sourcesAddedIntoSum[i]->getBusInterface()->getStateVarV()) << Trace::endline;
+        }
+        if (timeline != nullptr) {
+          constexpr size_t failingCriteriaNumber = 5;
+          const size_t maxFailingCritArraySize = std::min(failingCriteriaNumber, sourcesAddedIntoSum.size());
+          for (size_t loadIndex = 0; loadIndex < maxFailingCritArraySize; ++loadIndex) {
```

Comment:
If we take a look at the ticket only the criteria log should be given in the timeline and not the load details

Reply:
Isn't the criteria log equal to the load details ?